Users/abehenderson/CSX/Closures/hobbyTracker.js

```diff
@@ -0,0 +1,86 @@
+
+function hobbyTracker(hobbies) {
+  let cache = {};
+  for (let i = 0; i < hobbies.length; i++) {
+    cache[hobbies[i]] = 0;
+  }
+  
+  return function newFunc(hobby, hours) {
+    if (hobby === undefined && hours === undefined) {
+      for (const keys in cache) {
+        cache[keys] = 0;
+      }
+      return 'tracker has been reset!'
+    } else {
+      for (const property in cache) {
+         if (hobby === property) cache[property] += hours;
+      }
+    }
+    return cache;
```

Comment:
you only want to return the cache if hobby and hours are defined (see directions).

Reply:
This make sense, thanks